File: src/main/java/no/nav/foreldrepenger/boot/conditionals/EnvUtil.java
package no.nav.foreldrepenger.boot.conditionals;

import org.slf4j.Marker;
import org.slf4j.MarkerFactory;
import org.springframework.core.env.Environment;
import org.springframework.core.env.Profiles;

public final class EnvUtil {

    public static final Marker CONFIDENTIAL = MarkerFactory.getMarker("CONFIDENTIAL");

    private static final String FSS = "-fss";
    private static final String SBS = "-sbs";
    private static final String GCP = "-gcp";
    public static final String LOCAL = "local";
    public static final String DEFAULT = "default";

    public static final String TEST = "test";
    public static final String DEV = "dev";
    public static final String PROD = "prod";
    public static final String DEV_GCP = DEV + GCP;
    public static final String PROD_GCP = PROD + GCP;

    public static final String PROD_SBS = PROD + SBS;
    public static final String DEV_SBS = DEV + SBS;
    public static final String PROD_FSS = PROD + FSS;
    public static final String DEV_FSS = DEV + FSS;
    public static final String VTP = "vtp";

    private EnvUtil() {

    }

    public static boolean isDevOrLocal(Environment env) {
        return isLocal(env) || isDev(env);
    }

    public static boolean isVTP(Environment env) {
        return env.acceptsProfiles(Profiles.of(VTP));
    }

    public static boolean isDev(Environment env) {
        return env.acceptsProfiles(Profiles.of(DEV_SBS, DEV_GCP, DEV_FSS));
    }

    public static boolean isLocal(Environment env) {
        return (env == null) || env.acceptsProfiles(Profiles.of(LOCAL));
    }
}

File: src/main/kotlin/no/nav/boot/conditionals/EnvUtil.kt
package no.nav.boot.conditionals

import org.slf4j.MarkerFactory
import org.springframework.core.env.Environment
import org.springframework.core.env.Profiles

object EnvUtil {
    val CONFIDENTIAL = MarkerFactory.getMarker("CONFIDENTIAL")
    const val FSS = "fss"
    const val SBS = "sbs"
    const val LOCAL = "local"
    const val DEFAULT = "default"
    const val GCP = "gcp"
    const val TEST = "test"
    const val DEV = "dev"
    const val PROD = "prod"
    val DEV_GCP = join(DEV, GCP)
    val PROD_GCP = join(PROD, GCP)
    val PROD_SBS = join(PROD, SBS)
    val DEV_SBS = join(DEV, SBS)
    val PROD_FSS = join(PROD, FSS)
    val DEV_FSS = join(DEV, FSS)
    const val VTP = "vtp"
    private fun join(env: String, cluster: String) = "$env-$cluster"

    @JvmStatic
    fun isDevOrLocal(env: Environment) = isLocal(env) || isDev(env)

    @JvmStatic
    fun isVTP(env: Environment) = env.acceptsProfiles(Profiles.of(VTP))

    @JvmStatic
    fun isDev(env: Environment) = env.acceptsProfiles(Profiles.of(DEV_SBS, DEV_GCP, DEV_FSS))

    @JvmStatic
    fun isProd(env: Environment) = env.acceptsProfiles(Profiles.of(PROD_SBS, PROD_FSS,PROD_GCP))

    @JvmStatic
    fun isGcp(env: Environment) = env.acceptsProfiles(Profiles.of(DEV_GCP, PROD_GCP))

    @JvmStatic
    fun isLocal(env: Environment?) = env == null || env.acceptsProfiles(Profiles.of(LOCAL))

}